### ISSUE TITLE
Auto `versioning` from file hash

### ISSUE THREAD

**Issue by HarryBurns** [RealFaviconGenerator/gulp-real-favicon#5]
Hello again!

RealFaviconGenerator provides `versioning` future. It makes links in HTML looks like:

``` HTML
<link rel="manifest" href="manifest.json?ver=416270658374">
```

In config:

``` javascript
//...
versioning: {
    paramName: "ver",
    paramValue: "416270658374"
}
//...
```

It would be great to have `versioning: "auto"` option, that will calculate `?ver=...` automatically from file's hash.

This feature will be very handy :)

**Comment by HarryBurns** [comment]
The [gulp-hash](https://github.com/Dragory/gulp-hash) plugin could help.

**Comment by haydenbleasel** [comment]
@HarryBurns According to the [API docs](https://realfavicongenerator.net/api/non_interactive_api#.VmoXlBorLUI):

> When this section is set to true (ie. versioning: true), versioning is generated.

So just add `versioning: true` and it should be generated for you. If it doesn't work, then ask @phbernard haha. Piping to `gulp-hash` is also an option post-API-call but might be a hassle.

**Comment by HarryBurns** [comment]
@haydenbleasel According to API docs:

> When this section is set to true (ie. versioning: true), versioning is generated. The query parameter name is v and the value is a hashed timestamp.

I was talking about hash of file contents, not hashed timestamp :)

But, I think it would be better to move this issue to [favicons](https://github.com/haydenbleasel/favicons). Cause RFG web-service is more about manual generation, whilst [favicons](https://github.com/haydenbleasel/favicons) is about CI and automation of build process.

**Comment by haydenbleasel** [comment]
@HarryBurns That is true. Typically I'd say use `gulp-favicons` and pipe it to `gulp-hash` but I don't think you can easily affect the metadata in this way, so it may have to be integrated. Let's move this to a new issue on Favicons.

**Comment by phbernard** [comment]
Oh wow, I didn't see that one, sorry :(

This is already possible. Auto-everything:

```
//...
versioning: true
//...
```

Parameter name is `TheVersion` and auto-timestamp:

```
//...
versioning: {
    paramName: "TheVersion"
}
//...
```

**Comment by haydenbleasel** [comment]
@phbernard Yeah but he wants to hash by file contents rather than timestamp. Possible Feature Request for RFG.

**Comment by phbernard** [comment]
Oh, right, thank of for pointing this out.
